internal/advertisement-operator/config-watcher.go

```diff
@@ -4,15 +4,40 @@ import (
 	protocolv1 "github.com/liqoTech/liqo/api/advertisement-operator/v1"
 	policyv1 "github.com/liqoTech/liqo/api/cluster-config/v1"
 	"github.com/liqoTech/liqo/pkg/clusterConfig"
+	"github.com/liqoTech/liqo/pkg/crdClient"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/klog"
+	"time"
 )
 
-func (b *AdvertisementBroadcaster) WatchConfiguration(kubeconfigPath string) {
+func (b *AdvertisementBroadcaster) WatchConfiguration(kubeconfigPath string, client *crdClient.CRDClient) {
 	go clusterConfig.WatchConfiguration(func(configuration *policyv1.ClusterConfig) {
-		if configuration.Spec.AdvertisementConfig.ResourceSharingPercentage != b.ClusterConfig.ResourceSharingPercentage {
+		if !configuration.Spec.DiscoveryConfig.EnableAdvertisement {
```

Comment:
This config is to enable mDNS advertisement. I suggest to add another flag

I will change this name to EnableLANAdvertisement to avoid misunderstandings

Reply:
so at the moment we don't have a flag to enable/disable advertisement (i.e. the creation/deletion of the broadcaster)?